### PR TITLE
Add gRPC trailers to the RequestContext in AbstractUnsafeUnaryGrpcService and UnaryGrpcClient

### LINKED ISSUE #3724: Add trailers to the RequestContext in AbstractUnsafeUnaryGrpcService
See discussion in https://github.com/line/armeria/pull/3716#discussion_r676405645.

Trailers can be added to the `RequestContext` similarly to how it's done in [ArmeriaServerCall](https://github.com/line/armeria/blob/master/grpc/src/main/java/com/linecorp/armeria/server/grpc/ArmeriaServerCall.java#L383), so that [MeterIdPrefixFunction](https://github.com/line/armeria/blob/master/grpc/src/main/java/com/linecorp/armeria/common/grpc/GrpcMeterIdPrefixFunction.java#L108) can use the status.

### PR DESCRIPTION
**Pull request by alexc-db**
Motivation:

#3724 Trailers can be added to the `RequestContext` similarly to how it's done in [ArmeriaServerCall](https://github.com/line/armeria/blob/master/grpc/src/main/java/com/linecorp/armeria/server/grpc/ArmeriaServerCall.java#L383), so that [MeterIdPrefixFunction](https://github.com/line/armeria/blob/master/grpc/src/main/java/com/linecorp/armeria/common/grpc/GrpcMeterIdPrefixFunction.java#L108) can use the status.

Modifications:

- Add GrpcWebTrailers to the grpc-protocol package.
- Deprecate GrpcWebTrailers in the grpc package and delegate to the newly added class.
- Add calls to `GrpcWebTrailers.set` in the unary grcp server and client.

Result:

- Closes #3724.
- gRPC trailers are available in the RequestContext for unary gRPC server and client.

### CODE REVIEW

File: grpc/src/test/java/com/linecorp/armeria/client/grpc/protocol/UnaryGrpcClientTest.java [hunk 1]
```diff
@@ -92,13 +94,10 @@ void normal(SerializationFormat serializationFormat) throws Exception {
         try (ClientRequestContextCaptor captor = Clients.newContextCaptor()) {
             final byte[] responseBytes =
                     client.execute("/armeria.grpc.testing.TestService/UnaryCall", request.toByteArray()).join();
-            if (serializationFormat == UnaryGrpcSerializationFormats.PROTO) {
```
Comment (trustin) [inline grpc/src/test/java/com/linecorp/armeria/client/grpc/protocol/UnaryGrpcClientTest.java:95]: :tada:

**Review by trustin — APPROVED** [review]

**Review by ikhoon — APPROVED** [review]
Thanks! @alexc-db

**Review by minwoox — APPROVED** [review]
👍 👍 👍

**Pull request merged**